src/peft/tuners/lora/layer.py

```diff
@@ -111,6 +116,13 @@ def update_layer(self, adapter_name, r, lora_alpha, lora_dropout, init_lora_weig
                 else:
                     self.to(weight.device)
                 break
+
+        if use_dora:
+            self.dora_init(adapter_name)
+            self.use_dora[adapter_name] = True
```

Comment:
this technically means you can have multiple adapters, with some of them using DoRA and some not?

Reply:
I guess so, although it wouldn't currently be possible to configure it as such.

Reply:
even if you do `add_adapter` witha LoraConfig that has `use_dora=True` ?

Reply:
I guess we could have multiple adapters, some of which use DoRA and some don't. I believe this would be fine or can you think of an issue?